opus/application/static_media/js/opus.js

```diff
@@ -111,9 +112,9 @@ var opus = {
         whether to fire an ajax call.
         */
 
-        let selections = o_hash.getSelectionsFromHash();
+        let [selections, extras] = o_hash.getSelectionsExtrasFromHash();
 
-        if (selections === undefined) {
+        if (selections === undefined && extras === undefined) {
```

Comment:
Maybe this should be ||? If one of them is undefined, how the object comparison handle it? Not that this should ever actually happen.

Reply:
From getSelectionsExtrasFromHash, both selections and extras will be undefined if getHash return an empty string. There won't be a case that only of them is undefined. If one of them doesn't exist in url, it will be an empty object, like [{"planet": ["mars", "neptune"]}, {}] or [{}, {"qtype-time": ["any"]}]

Reply:
Right...so it's confusing that you are checking for *both* to be true, since it's impossible for one to be true without the other. With any other language, what I'd want to do instead is:
```
if (selections === undefined) {
    assert(extras === undefined);
    ....
}
```
to catch latent or new bugs, but JavaScript doesn't have anything like an assert function. Perhaps we could still do something like this, but instead use `console.error()` to log an error message saying that there was a problem with extras.